File: chatbot-backend/src/backend/models/schedule.py
from odmantic import Model
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional

# Model lưu trữ trên MongoDB (ODMantic)
class Schedule(Model):
    user: str  
    subject: str = Field(..., max_length=100)  # Giới hạn 100 ký tự
    start_time: datetime
    end_time: datetime
    category_color: Optional[str] = Field(default="#0000FF")

    class Config:
        collection = "schedules"  

# Schema dùng để validate request/response (Pydantic)
class ScheduleSchema(BaseModel):
    user: str
    subject: str = Field(..., max_length=100)
    start_time: datetime
    end_time: datetime
    category_color: Optional[str] = None

File: backend/model/Schedule.js
const mongoose = require('mongoose');

const ScheduleSchema = new mongoose.Schema(
  {
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    subject: { type: String, required: true, trim: true, maxlength: 100 }, 
    startTime: { type: Date, required: true },
    endTime: { type: Date, required: true },
    categoryColor: { type: String, required: true, default: '#0000FF' } 
  },
  { timestamps: true } 
);

module.exports = mongoose.model('Schedule', ScheduleSchema);
